core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -889,21 +891,40 @@ private class PythonAccumulatorParam(@transient private val serverHost: String,
     socket
   }
 
-  override def zero(value: JList[Array[Byte]]): JList[Array[Byte]] = new JArrayList
+  override def reset(): Unit = {
+    this._acc = new JArrayList[Array[Byte]]
+  }
+
+  override def isZero: Boolean = {
+    this._acc.isEmpty
+  }
 
-  override def addInPlace(val1: JList[Array[Byte]], val2: JList[Array[Byte]])
-      : JList[Array[Byte]] = synchronized {
+  override def copy(): PythonAccumulatorV2 = {
+    val newAcc = new PythonAccumulatorV2(serverHost, serverPort)
+    newAcc._acc = this._acc
```

Comment:
OK but this keeps the original accumulator's data -- it even makes both point to the same instance of the data. I suspect I'm missing something.